frontend/public/components/deploy-image.tsx

```diff
@@ -0,0 +1,412 @@
+  return !user ||
+          user === '0' ||
+          user === 'root';
+};
+
+const getImagePorts = image => _.get(image, 'dockerImageMetadata.Config.ExposedPorts') || _.get(image, 'image.dockerImageMetadata.ContainerConfig.ExposedPorts');
+
+const parsePorts = image => {
+  // map ports to k8s structure
+  const parsePortsFromSpec = portSpec => {
+    const ports = [];
+    _.each(portSpec, (value, key) => {
+      const parts = key.split('/');
+      if (parts.length === 1){
+        parts.push('tcp');
+      }
+
+      const containerPort = parseInt(parts[0], 10);
+      if (!_.isFinite(containerPort)) {
+        // eslint-disable-next-line no-console
+        console.log(`Container port ${parts[0]} is not a number for image ${image.metadata.name}`);
+      } else {
+        ports.push({
+          containerPort: containerPort,
+          protocol: parts[1].toUpperCase()
+        });
+      }
+    });
+
+    return _.sortBy(ports, 'containerPort');
+  };
+
+  return parsePortsFromSpec(getImagePorts(image) || []);
+};
+
+const ImagePorts = ({ports, name}) => <React.Fragment>
+  {_.size(ports) > 1 ? 'Ports ' : 'Port '}
+  {_.map(ports, (value, key) => key.toUpperCase()).join(', ')} will be load balanced by Service <strong>{name || '<name>'}</strong>.
+  <div>Other containers can access this service through the hostname <strong>{name || '<name>'}</strong>.</div>
+</React.Fragment>;
+
+export class DeployImage extends React.Component<DeployImageProps, DeployImageState> {
+  constructor (props) {
+    super(props);
+
+    this.state = {
+      namespace: '',
+      imageName: '',
+      loading: false,
+      inProgress: false,
+      name: '',
+    };
+  }
+
+  onNamespaceChange = (namespace: string) => {
+    this.setState({namespace});
+  };
+
+  onImageNameChange: React.ReactEventHandler<HTMLInputElement> = event => {
+    this.setState({imageName: event.currentTarget.value});
+  }
+
+  onKeyPress = event => {
+    if (this.state.imageName && event.key === 'Enter') {
+      this.search(event);
+    }
+  }
+
+  onNameChange: React.ReactEventHandler<HTMLInputElement> = event => {
+    this.setState({name: event.currentTarget.value});
+  };
+
+  search = event => {
+    event.preventDefault();
+
+    const { namespace, imageName } = this.state;
+    const importImage = {
+      kind: 'ImageStreamImport',
+      apiVersion: 'image.openshift.io/v1',
+      metadata: {
+        name: imageName,
+        namespace
+      },
+      spec: {
+        import: false,
+        images: [{
+          from: {
+            kind: 'DockerImage',
+            name: imageName
+          }
+        }]
+      },
+      status: {}
+    };
+
+    this.setState({
+      loading: true,
+      isi: null,
+      name: '',
+      error: null,
+      searchError: null,
+    });
+
+    k8sCreate(ImageStreamImportsModel, importImage)
+      .then((imageStreamImport) => {
+        const name = _.get(imageStreamImport, 'spec.images[0].from.name'),
```

Comment:
nit: this should probably be moved inside the if since it's only used there.

(missed on last review, sorry)